### PR TITLE
Prevent new user autofill

### PR DESCRIPTION
**Pull request by rbf**
Connected to #476

I've added three mechanisms to prevent wrong usernames in the self-registration form:

1. Added `autocomplete="off"` at the `form` element (not all browsers honor that though)
1. Added following attributes to `new username` field to enhance the auto-fill prevention:

    ```
    autocomplete="off"
    readonly
    onfocus="this.removeAttribute('readonly');"
    ```

1. (Re)trigger form validation before sending the request.

### CODE REVIEW

**Comment by mebster** [comment]
When deploying Drupal app, in the Run... Dialog, the email got replaced by a non-email value on Safari (didn't try other browsers). Will this fix also address this issue? The result was a silently failed deployment.  

_________
Marc-Elian Bégin
CEO, Co-Founder
SixSq
Geneva, Switzerland
meb@sixsq.com
+41 77 44 68 119
@lemeb
@sixsq
sixsq.com

File: clj/resources/static_content/js/login.js [hunk 1]
```diff
@@ -7,5 +7,11 @@ jQuery( function() { ( function( $$, $, undefined ) {
             "Sorry, something unexpected happend while processing your registration request.")
```
Comment (loomis) [inline clj/resources/static_content/js/login.js]: "happend" --> "happened"

File: clj/resources/static_content/js/login.js [hunk 2]
```diff
@@ -7,5 +7,11 @@ jQuery( function() { ( function( $$, $, undefined ) {
...
         .onSuccessAlert("Registration successful",
             "You will receive in short an email with instructions to validate the account.")
```
Comment (loomis) [inline clj/resources/static_content/js/login.js]: "receive in short an email" --> "shortly receive an email"

**Comment by loomis** [comment]
This PR only affects the login page and javascript.  I don't believe it will have any affect on the run dialog.  Another issue should be opened for that.

**Comment by rbf** [comment]
> This PR only affects the login page and javascript. I don't believe it will have any affect on the run dialog. Another issue should be opened for that.

I confirm.

Thanks for the merge @loomis.

**Pull request merged**